CMakeLists.txt

```diff
@@ -153,6 +153,46 @@ if(ITKPythonPackage_SUPERBUILD)
   message(STATUS "SuperBuild - Searching for python[OK]")
   message(STATUS "SuperBuild -   DOXYGEN_EXECUTABLE: ${DOXYGEN_EXECUTABLE}")
 
+  # CMake configuration variables to pass to ITK's build
+  set(ep_itk_cmake_cache_args "")
+  foreach(var ITK_WRAP_DOC
+    BUILD_SHARED_LIBS
+    ITK_BUILD_DEFAULT_MODULES
+    ITKGroup_Core
+    ITKGroup_Filtering
+    ITKGroup_IO
+    ITKGroup_Nonunit
+    ITKGroup_Numerics
+    ITKGroup_Registration
+    ITKGroup_Segmentation
+    ITKGroup_Video
+    ITK_WRAP_IMAGE_DIMS
+    ITK_WRAP_VECTOR_COMPONENTS
+    ITK_WRAP_complex_double
+    ITK_WRAP_complex_float
+    ITK_WRAP_covariant_vector_double
+    ITK_WRAP_covariant_vector_float
+    ITK_WRAP_double
+    ITK_WRAP_float
+    ITK_WRAP_rgb_unsigned_char
+    ITK_WRAP_rgb_unsigned_short
+    ITK_WRAP_rgba_unsigned_char
+    ITK_WRAP_rgba_unsigned_short
+    ITK_WRAP_signed_char
+    ITK_WRAP_signed_long_long
+    ITK_WRAP_signed_short
+    ITK_WRAP_unsigned_char
+    ITK_WRAP_unsigned_long_long
+    ITK_WRAP_unsigned_short
+    ITK_WRAP_vector_double
+    ITK_WRAP_vector_float
+    )
+    if(DEFINED ${var})
+      list(APPEND ep_itk_cmake_cache_args "-D${var}=${${var}}")
+    endif()
+  endforeach()
+  # Todo, also pass all Module_* variables
+  message(STATUS "ITK CMake Cache Args -   ${ep_itk_cmake_cache_args}")
```

Comment:
You could simplify this by using https://cmake.org/cmake/help/v3.11/prop_dir/CACHE_VARIABLES.html

You could then automatically propagate all variables prefixed with either `ITK_WRAP_` or `Modules_` prefix

Reply:
@jcfr The documentation says that this property is intended for debug purposes. Do you think it is a good idea to use it in this case?

Reply:
> this property is intended for debug purposes

This is a hint. It makes sense to use it here.

We should submit a MR to cmake to change this too.

With something like `this property is useful for debug purposes`

Other references:
* http://cmake.3232098.n2.nabble.com/Relaying-all-command-line-arguments-from-SuperBuild-to-ExternalProject-Add-tp7597038p7597045.html
* https://github.com/search?l=CMake&q=CACHE_VARIABLES&type=Code&utf8=%E2%9C%93

Reply:
Great. Thanks for your input.